src/ring/adapter/jetty9.clj

```diff
@@ -250,20 +255,36 @@
                                           job-queue)
                    (.setDaemon daemon?)))
         server (doto (Server. ^ThreadPool pool)
-                 (.addBean (ScheduledExecutorScheduler.)))
+                 (.addBean (ScheduledExecutorScheduler.))
+                 (.addBean (proxy [AbstractLifeCycle] []
+                             (doStart [] (when-some [f (:lifecycle-start options)] (f)))
+                             (doStop  [] (when-some [f (:lifecycle-end options)] (f)))))
+                 (.setStopAtShutdown true))
         http-configuration (http-config options)
         ssl? (or ssl? ssl-port)
         ssl-port (or ssl-port (when ssl? 443))
-        ssl-factory (ssl-context-factory options)
+        ssl-factory (delay (ssl-context-factory options)) ;; might end up not needing it
```

Comment:
Do we still need to `delay` it?

Reply:
By `might end up not needing it`, I mean the ssl-context-factory object - i.e. caller doesn't want `ssl?`, nor `http3`. That's why it's behind a delay.